Index: linux/arch/x86/include/asm/ptrace.h
===================================================================
--- linux.orig/arch/x86/include/asm/ptrace.h
+++ linux/arch/x86/include/asm/ptrace.h
@@ -143,8 +143,9 @@ extern void send_sigtrap(struct task_str
 			 int error_code, int si_code);
 void signal_fault(struct pt_regs *regs, void __user *frame, char *where);
 
-extern long syscall_trace_enter(struct pt_regs *);
-extern void syscall_trace_leave(struct pt_regs *);
+#define asmregparm __attribute__((regparm(3)))
+asmregparm extern long syscall_trace_enter(struct pt_regs *);
+asmregparm extern void syscall_trace_leave(struct pt_regs *);
 
 static inline unsigned long regs_return_value(struct pt_regs *regs)
 {
